CEGameApp.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CEGameApp.API.Data;
+using CEGameApp.API.Dtos;
+using CEGameApp.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace CEGameApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthRepository _repo;
+        private readonly IConfiguration _config;
+
+        public AuthController(IAuthRepository repo, IConfiguration config)
+        {
+            _repo = repo;
+            _config = config;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserForRegisterDto userForRegisterDto)
+        {
+            //validate request
+            userForRegisterDto.UserName = userForRegisterDto.UserName.ToLower();
+            
+            if (await _repo.UserExists(userForRegisterDto.UserName))
+                return BadRequest("Username already exists");
+            
+            var userToCreate = new Users
+            {
+                Name = userForRegisterDto.UserName
+            };
+
+            var createdUser = await _repo.Register(userToCreate, userForRegisterDto.Password);
+
+            return StatusCode(201);  //will change to createdatroute once we have a place to get data from
+        }
+
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(UserForLoginDto userForLoginDto)
+        {
+            var userFromRepo = await _repo.Login(userForLoginDto.UserName.ToLower(), userForLoginDto.Password);
+
+            if (userFromRepo == null)
+                return Unauthorized();
+
+            //***************  All of this to create a token to send to users logging in *******************
+            
+            //2 claims being sent back as part of token containing user Id and username
```

Comment:
two things here.  so you have comments explaining what it is your are doing and why you are doing.  I would say this is a code smell because it is clear what is happning from just reading the next lines of code.  but if your counter argument is "theres a bunch happening here" then I would agree, there is a bunch happening here.  

like the previous comments above, this logic can exist in an ILoginService contracted class.  this class will be responsible for handling all the logic needed to create a login.  in other words, keep your action methods AS LEAN AS POSSIBLE.  you are doing too much.

Reply:
Done.